noir-projects/noir-protocol-circuits/crates/types/src/merkle_tree/indexed_tree/check_valid_low_leaf.nr

```diff
@@ -0,0 +1,78 @@
+use crate::merkle_tree::leaf_preimage::IndexedTreeLeafPreimage;
+
+pub fn assert_check_valid_low_leaf<LEAF_PREIMAGE>(
+    key: Field,
+    low_leaf_preimage: LEAF_PREIMAGE
+) where LEAF_PREIMAGE: IndexedTreeLeafPreimage {
+    let low_key = low_leaf_preimage.get_key();
+    let next_key = low_leaf_preimage.get_next_key();
+
```

Comment:
Removed the check:
```
let is_empty_leaf = (low_key == 0) & (next_key == 0);
assert(!is_empty_leaf);
```

Reason is added to a comment below with the test `test_assert_check_empty_low_leaf`:
// An all-zero low leaf should be valid. It could be used as the first dummy leaf in a tree.
// It's not possible to prove against an empty leaf at an uninitialized index.
// The membership check will fail because the leaf value hash(0, 0) is not 0.

Reply:
Yeah the `prefilledSize` is greater than 1 in real world so there won't be an all-zero leaf. But its default value is 1, so we can still consider the possibility here that there might just be 1 all-zero leaf in the tree. And this assumption is not adding more constraints so we should be good 😂